src/itree.cpp

```diff
@@ -122,6 +122,19 @@ int main(int argc, char** argv){
         auto path = correct_path(vm, env);
         irods::experimental::client_connection conn;
 
+        const auto s = fs::client::status(conn, path);
+
+        if (!fs::client::exists(s))
+        {
+            std::cerr << fmt::format("Error: logical path [{}] does not exist.",path.c_str()) << std::endl;
+            return 1;
+        }
+        else if (fs::client::is_data_object(s))
+        {
```

Comment:
This leaves room for things other than collections to trigger errors.

Instead, you should check for a collection explicitly. For example:
```c++
if (!fs::client::is_collection(s)) {
```
_Notice the opening curly brace has bee moved to the end of the line as well._

If the other if-stmt is removed, this can be rewritten as:
```c++
if (!fs::client::is_collection(conn, path)) {
```
And the call to `fs::client::status(conn, path)` can be removed too.

Reply:
done.